Choreographer_30FPS/app/build.gradle

```diff
@@ -0,0 +1,64 @@
+apply plugin: 'com.android.model.application'
+
+model {
+    android {
+        compileSdkVersion = 23
+        buildToolsVersion = "23.0.2"
+
+        defaultConfig.with {
+            applicationId = "com.sample.choreographer"
+            minSdkVersion.apiLevel = 16
+            targetSdkVersion.apiLevel = 23
+        }
+
+    }
+
+    android.ndk {
+        moduleName = "ChoreographerNativeActivity"
+        cppFlags.addAll(["-std=c++11",
+                         "-I${file("src/main/jni/Common/native_app_glue")}".toString(),
+                         "-I${file("src/main/jni/Common/ndk_helper")}".toString()])
+        ldLibs.addAll(["android", "EGL", "GLESv2", "dl", "log", "atomic"])
+        stl     = "gnustl_static"
+    }
+    // jni is the default dir; config this if yours is in different directory
+    android.sources {
+        main {
+            jni {
+                source {
+                    srcDirs 'src/main/jni'
+                    srcDirs '../../Common'
+                }
+            }
+        }
+    }
+    android.buildTypes {
+        release {
+            minifyEnabled = false
+            proguardFiles.add(file('proguard-rules.txt'))
+        }
+    }
+    // Turn on hard float support in armeabi-v7a
+    android.abis {
+        create("armeabi-v7a") {
+            cppFlags.addAll(["-mhard-float", "-D_NDK_MATH_NO_SOFTFP=1", "-mfloat-abi=hard"])
+            ldLibs.add("m_hard")
+            ldFlags.add("-Wl,--no-warn-mismatch")
+        }
+    }
+    android.productFlavors {
+        create ("arm7") {
+            ndk.abiFilters.add("armeabi-v7a")
+        }
+        create ("arm8") {
+            ndk.abiFilters.add("arm64-v8a")
```

Comment:
A few things:
1) tuck all things that having a android.*  into android{} closure, we are moving away from this syntax
2) like to dump the *.toString() style as much as possible, the syntax is clumsy. this could be changed from:
"-I${file("src/main/jni/Common/native_app_glue")}".toString(),
to:
'-I' + file(file('src/main/jni/Common/native_app_glue'),
3) use single quotation as much as possible:
    if there is no variable expansion needed, use single quotation
    if there IS expansion necessary [like ${nfkPath}], yeah we have to use double quotation

the idea is to make build script short, simple, and small: C/C++ code is what makes us proud and avoid too much script in the way